HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -669,6 +694,20 @@ def check_for_errors
         end
       end
 
+      if (not @dst_begin_month.nil?) && (not @dst_end_month.nil?)
+        if @dst_begin_month > @dst_end_month
+          fail "Daylight Saving Begin Month (#{@dst_begin_month}) cannot come after Daylight Saving End Month (#{@dst_end_month})."
+        end
+
+        if (not @dst_begin_day_of_month.nil?) && (not @dst_end_day_of_month.nil?)
+          if @dst_begin_month == @dst_end_month
+            if @dst_begin_day_of_month > @dst_end_day_of_month
+              fail "Daylight Saving Begin Day of Month (#{@dst_begin_day_of_month}) cannot come after Daylight Saving End Day of Month (#{@dst_end_day_of_month}) for the same month (#{@dst_begin_month})."
+            end
+          end
+        end
+      end
+
```

Comment:
I thought you were intentionally excluding this in your first commit and almost gave it a 👍. I don't think these checks should exist because DST will wrap around the year in the southern hemisphere. I assume EnergyPlus will allow it but you can double-check.

Reply:
I was intentionally excluding at first. But then I checked if you can wrap around the year- and got "Simulation Unsuccessful".

Reply:
And did the error message say?

The documentation examples show that it should be supported:
https://bigladdersoftware.com/epx/docs/9-3/input-output-reference/group-location-climate-weather-file-access.html#runperiodcontroldaylightsavingtime

Reply:
I copied the example from the link above, for when dst start is after dst end, and I get an energyplys segmentation fault.

Reply:
Well, I think we should allow our code to support it since E+ documentation suggests it should work. Can you create an E+ GitHub issue on the crash and attach your IDF?

Reply:
I updated the code to allow wrap around daylight saving time period (even though it appears there is currently an energyplus issue with that).